crates/solver/src/s3_instance_upload.rs

```diff
@@ -10,36 +10,48 @@ use {
 
 #[derive(Default)]
 pub struct Config {
-    pub region: String,
     pub bucket: String,
-    pub access_key_id: String,
-    pub secret_access_key: String,
     /// Prepended to the auction id to form the final filename. Something like
     /// "staging/mainnet/quasimodo/". Should end with `/` if intended to be a
     /// folder.
     pub filename_prefix: String,
 }
 
+#[derive(Default)]
+pub struct Credentials {
+    pub access_key_id: String,
+    pub secret_access_key: String,
+    pub region: String,
+}
+
 pub struct S3InstanceUploader {
     bucket: String,
     filename_prefix: String,
     client: Client,
 }
 
 impl S3InstanceUploader {
-    pub fn new(config: Config) -> Self {
-        let aws_config = aws_types::sdk_config::Builder::default()
-            .region(Region::new(config.region))
-            .credentials_provider(SharedCredentialsProvider::new(Credentials::from_keys(
-                config.access_key_id,
-                config.secret_access_key,
-                None,
-            )))
-            .build();
+    pub async fn aws_credentials_from_cli_or_env(cli: Option<Credentials>) -> aws_types::SdkConfig {
+        match cli {
+            Some(args) => {
+                aws_types::sdk_config::Builder::default()
+                    .region(Region::new(args.region))
+                    .credentials_provider(SharedCredentialsProvider::new(
+                        AwsCredentials::from_keys(args.access_key_id, args.secret_access_key, None),
+                    ))
+                    .build()
+            }
+            // According to the AWS docs this is the recommended way to use the SDK. Unfortunately
+            // we don't have a way to detect errors when loading from the environment.
+            None => aws_config::load_from_env().await,
```

Comment:
I believe this will just panic, so as long as we call this on startup we should get notified if something is wrong with the configuration. Maybe we could add the comment that in order to avoid the panic you should not specify a `s3_upload` parameter,

Reply:
Unfortunately this is not the case. I could do a sanity check in the constructor that it's able to upload a test file and panic if this is not the case.
WDYT?